apps/sms/services/js/conversation/conversation_service.js

```diff
@@ -22,26 +38,74 @@ if (!('BridgeServiceMixin' in self)) {
   ]);
 
   var ConversationService = {
+    mobileMessageClient: null,
+
     /**
      * Initializes our service using the Service mixin's initService.
      */
     init() {
       this.initService();
+
+      this.mobileMessageClient = bridge.client(
+        'mozMobileMessageShim',
+        new BroadcastChannel('mobile-message-shim-channel')
+      ).plugin(streamClient);
     },
 
     /**
      * Stream that returns everything needed to display conversations in the
      * inbox view.
-     * @param {ServiceStream.<ConversationSummary>} The stream to use in the
-     * implementation: it will be passed autoamtically by the Bridge library.
+     * @param {ServiceStream.<ConversationSummary>} serviceStream The stream to
+     * use in the implementation: it will be passed automatically by the Bridge
+     * library.
      */
-    getAllConversations(stream) {},
+    getAllConversations(serviceStream) {
+      var getThreadsStream = this.mobileMessageClient.stream('getThreads');
```

Comment:
> Mmm, I really doubt there is any significant difference if I change the order of Drafts.request and client.stream. Am I missing something?

My thinking here is that both the stream request and the draft request start a IDB request in a separate thread (but for the stream request, it would happen after the shim gets the message, so later). So the request can go on even if there are things happening on this thread. I don't know if this gives any performance different though.